modules/dokkatoo-plugin/src/main/kotlin/adapters/DokkatooKotlinAdapter.kt

```diff
@@ -266,51 +262,25 @@ private class KotlinCompilationDetailsBuilder(
   private fun collectKotlinCompilationClasspath(
     compilation: KotlinCompilation<*>,
   ): FileCollection {
-
     val compilationClasspath = objects.fileCollection()
-    fun collectConfiguration(named: String) {
-      configurations.collectIncomingFiles(named = named, collector = compilationClasspath)
-
-      // need to fetch JAVA_RUNTIME files explicitly, because Android Gradle Plugin is weird and
-      // doesn't seem to register the attributes explicitly on its configurations
-      @Suppress("UnstableApiUsage")
-      configurations.collectIncomingFiles(named = named, collector = compilationClasspath) {
-        withVariantReselection()
-        attributes {
-          attribute(USAGE_ATTRIBUTE, objects.named(JAVA_API))
-        }
-        lenient(true)
+    compilationClasspath.from(compilation.compileDependencyFiles)
+
+    if (kotlinGradlePluginVersion != null && kotlinGradlePluginVersion <= KotlinVersion(1, 9, 255)) {
+      if (compilation is AbstractKotlinNativeCompilation) {
+        val konanDistribution = KotlinNativeDistributionAccessor(compilation.target.project)
+        // KT-61559: In Kotlin 2.0 this will be part of [compilation.compileDependencyFiles]
+        compilationClasspath.from(konanDistribution.stdlibDir)
+        compilationClasspath.from(konanDistribution.platformDependencies(compilation.konanTarget))
       }
     }
-
-    val standardConfigurations = buildSet {
-      addAll(compilation.relatedConfigurationNames)
-      addAll(compilation.kotlinSourceSets.flatMap { it.relatedConfigurationNames })
-    }
-
-    logger.info("[$projectPath] compilation ${compilation.name} has ${standardConfigurations.size} standard configurations $standardConfigurations")
-
-    standardConfigurations.forEach { collectConfiguration(it) }
-
-    if (compilation is AbstractKotlinNativeCompilation) {
-      // K/N doesn't correctly set task dependencies, the configuration
-      // `defaultSourceSet.implementationMetadataConfigurationName`
-      // will trigger a bunch of Gradle warnings about "using file outputs without task dependencies",
-      // so K/N compilations need to explicitly depend on the compilation tasks
-      // UPDATE: actually I think is wrong, it's a bug with the K/N 'commonize for IDE' tasks
-      // see: https://github.com/Kotlin/dokka/issues/2977
-      collectConfiguration(
-        named = compilation.defaultSourceSet.implementationMetadataConfigurationName,
-//          builtBy = kotlinCompilation.compileKotlinTaskProvider
-      )
-    }
-
     return compilationClasspath
   }
 
   companion object {
     private fun KotlinCompilation<*>.isMain(): Boolean {
```

Comment:
```suggestion
    private fun KotlinCompilation<*>.isPublishable(): Boolean {
```

- [ ] TODO rename all 'isMain' stuff to 'isPublishable'